### PR TITLE
⬆️ Upgrade Starlette to >=0.37.2,<0.38.0, remove Starlette filterwarning for internal tests

### PR DESCRIPTION
**Pull request by nothielf**
Upgrading Starlette should solve
```
DeprecationWarning: The 'app' shortcut is now deprecated. Use the explicit style 'transport=WSGITransport(app=...)' instead.                                                                                                               
```
When running tests using TestClient

Also, removed filterwarning for method parameter. That was removed on Starlette [0.35.1](https://github.com/encode/starlette/releases/tag/0.35.1)

### CODE REVIEW

**Comment by flying-sheep** [comment]
should be 0.37.1 because of a bug in starlette 0.37.0 I think.

File: pyproject.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ classifiers = [
     "Topic :: Internet :: WWW/HTTP",
 ]
 dependencies = [
-    "starlette>=0.36.3,<0.37.0",
+    "starlette>=0.37.0,<0.38.0",
```
Comment (alfreedom) [inline pyproject.toml:44]: use 0.37.2,<0.38.0

**Review by alfreedom — CHANGES_REQUESTED** [review]

**Comment by ar090** [comment]
Thanks for putting this together! Would love to see this merged as it also fixes the issue discussed [here](https://github.com/tiangolo/fastapi/discussions/11019)

File: pyproject.toml (hunk 1 above)
Comment (nothielf) [reply pyproject.toml]: done!

**Comment by ar090** [comment]
I noticed theres a few other PR's sprouting up that are attempting to make similar changes:
https://github.com/tiangolo/fastapi/pull/11315
https://github.com/tiangolo/fastapi/pull/11320
@alfreedom any chance you would be available to ✅? 🙏

**Review by alfreedom — APPROVED** [review]

**Comment by alfreedom** [comment]
> I noticed theres a few other PR's sprouting up that are attempting to make similar changes: #11315 #11320 @alfreedom any chance you would be available to ✅ this one? 🙏

Approved, sorry for the delay.

**Comment by ar090** [comment]
thanks for the help here @alfreedom! @nothielf are we good to merge or waiting on anything else? Sorry to keep bumping this, we are being forced to upgrade for a vulnerability that comes with deadlines from compliance.

**Comment by nothielf** [comment]
i dont have write access to merge. but it seems is good to go  :)

**Comment by nothielf** [comment]
@alfreedom  are you able to merge it?

**Comment by alfreedom** [comment]
> @alfreedom are you able to merge it?

Nope, just maintainers have write access, maybe @alejsdev

File: pyproject.toml [hunk 2]
```diff
@@ -41,7 +41,7 @@ classifiers = [
     "Topic :: Internet :: WWW/HTTP",
 ]
 dependencies = [
-    "starlette>=0.36.3,<0.37.0",
+    "starlette>=0.37.2,<0.38.0",
```
Comment (Kludex) [inline pyproject.toml:44]: This should be good?
```suggestion
    "starlette>=0.36.3,<0.38.0",
```

**Review by Kludex — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (tvasenin) [reply pyproject.toml:44]: No, minimal version should NOT be less than `0.37.2`, as it fixes `DeprecationWarning` described in the first post
Also, `0.37.1` fixes another, very important regression: see #11320.

**Comment by ar090** [comment]
@Kludex sorry to bug you on this but is this something you can merge by chance? Not sure who has the ability to merge for FastAPI

**Comment by andgoldin** [comment]
I would like to see this get merged soon as well. Like others, I'm trying to update FastAPI to resolve [CVE-2024-24762](https://github.com/advisories/GHSA-2jv5-9r88-3w3p), but am running into the `starlette` issue discussed in #11019 when trying to do so.

**Comment by Kludex** [comment]
You can just bump Python-multipart.

**Comment by aswanidutt87** [comment]
> Only @tiangolo .

@tiangolo , please merge this change as we are stuck with a CVE and are keep skipping it from sometime now.
Thank you.

**Comment by tiangolo** [comment]
Great, thanks for the contribution @nothielf! :rocket: 

And thanks all for the comments. :coffee: 

This will be available in FastAPI `0.110.1`, released in the next hours. :tada:

**Pull request merged**